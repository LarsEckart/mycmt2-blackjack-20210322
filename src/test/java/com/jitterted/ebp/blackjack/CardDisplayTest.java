package com.jitterted.ebp.blackjack;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class CardDisplayTest {

  @Test
  void displayTenAsString() {
    Card card = new Card(Suit.HEARTS, Rank.TEN);

    Approvals.verify(ConsoleCard.display(card));
  }

  @Test
  void displayNonTenAsString() {
    Card card = new Card(Suit.CLUBS, Rank.SEVEN);

    Approvals.verify(ConsoleCard.display(card));
  }

  @Test
  void all_combinations() {
    CombinationApprovals.verifyAllCombinations(this::displayCard, Suit.values(), Rank.values());
  }

  private String displayCard(Suit suit, Rank rank) {
    return ConsoleCard.display(new Card(suit, rank));
  }
}
