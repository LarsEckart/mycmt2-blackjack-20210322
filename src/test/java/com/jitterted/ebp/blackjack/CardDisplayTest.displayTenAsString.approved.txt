[31m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘